sdk/metric/aggregation/aggregation.go

```diff
@@ -0,0 +1,152 @@
+
+var _ Aggregation = Drop{}
+
+func (Drop) private() {}
+
+// Err returns an error for any misconfiguration. A Drop aggregation has no
+// parameters and cannot be misconfigured, therefore this always returns nil.
+func (Drop) Err() error { return nil }
+
+// Default is an aggregation that uses the default instrument kind selection
+// mapping to select another aggregation. A metric reader can be configured to
+// make an aggregation selection based on instrument kind that differs from
+// the default selection mapping. This aggregation ensures that the default
+// instrument kind mapping. The default instrument kind mapping uses the
+// following selection: Counter ⇨ Sum, Asynchronous Counter ⇨ Sum,
+// UpDownCounter ⇨ Sum, Asynchronous UpDownCounter ⇨ Sum, Asynchronous Gauge ⇨
+// LastValue, Histogram ⇨ ExplicitBucketHistogram.
+type Default struct{} // Default has no parameters.
+
+var _ Aggregation = Default{}
+
+func (Default) private() {}
+
+// Err returns an error for any misconfiguration. A Default aggregation has no
+// parameters and cannot be misconfigured, therefore this always returns nil.
+func (Default) Err() error { return nil }
+
+// Sum is an aggregation that summarizes a set of measurements as their
+// arithmetic sum.
+type Sum struct{} // Sum has no parameters.
+
+var _ Aggregation = Sum{}
+
+func (Sum) private() {}
+
+// Err returns an error for any misconfiguration. A Sum aggregation has no
+// parameters and cannot be misconfigured, therefore this always returns nil.
+func (Sum) Err() error { return nil }
+
+// LastValue is an aggregation that summarizes a set of measurements as the
+// last one made.
+type LastValue struct{} // LastValue has no parameters.
+
+var _ Aggregation = LastValue{}
+
+func (LastValue) private() {}
+
+// Err returns an error for any misconfiguration. A LastValue aggregation has
+// no parameters and cannot be misconfigured, therefore this always returns
+// nil.
+func (LastValue) Err() error { return nil }
+
+// ExplicitBucketHistogram is an aggregation that summarizes a set of
+// measurements as an histogram with explicitly defined buckets.
+type ExplicitBucketHistogram struct {
+	// Boundaries are the increasing bucket boundary values. Boundary values
+	// define bucket upper bounds. Buckets are exclusive of their lower
+	// boundary and inclusive of their upper bound (except at positive
+	// infinity). A measurement is defined to fall into the greatest-numbered
+	// bucket with a boundary that is greater than or equal to the
+	// measurement. As an example, boundaries defined as:
+	//
+	// []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 1000}
+	//
+	// Will define these buckets:
+	//
+	// (-∞, 0], (0, 5.0], (5.0, 10.0], (10.0, 25.0], (25.0, 50.0],
+	// (50.0, 75.0], (75.0, 100.0], (100.0, 250.0], (250.0, 500.0],
+	// (500.0, 1000.0], (1000.0, +∞)
+	Boundaries []float64
+	// RecordMinMax indicates whether to record the min and max of the
+	// distribution.
+	RecordMinMax bool
```

Comment:
Do we need to find a way to make this default to true? 

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation

Reply:
I wasn't completely sure of that. The way I read it was the default is the aggregation returned from the default aggregation selector (i.e. when the user does nothing), that is the one that correctly sets the default boundaries and `RecordMinMax`.

Reply:
I'll update to `NoMinMax`

Reply:
Changed.